feat: Add Middle East map

abdyar opened the pull request:
## Description:

Adds a new playable map of the Middle East. This map includes key nations and regions from the area, complete with unique flags.

The following files for the map are included in the pull request:

Middleeast.bin (Binary map data)

Middleeast.json (Map metadata and nation data)

Middleeast.jpg (Main map image)

MiddleeastMini.bin (Minimap data)

MiddleeastThumb.webp (Map thumbnail)

All files have been placed in the resources/ directory under a new folder named middle-east-map.


## Please complete the following:

- [x] I have added screenshots for all UI updates
- [ ] I process any text displayed to the user through translateText() and I've added it to the en.json file
- [ ] I have added relevant tests to the test directory
- [x] I confirm I have thoroughly tested these changes and take full responsibility for any bugs introduced

## Please put your Discord username so you can be contacted if a bug or regression is found:

DISCORD_USERNAME : abdulrahman3148

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/openfrontio/OpenFrontIO?pullRequest=1989) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/openfrontio/OpenFrontIO?pullRequest=1989) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/openfrontio/OpenFrontIO?pullRequest=1989) it.</sub>

TheGiraffe3 commented on src/server/MapPlaylist.ts:
```diff
@@ -47,6 +47,7 @@ const frequency: Partial<Record<GameMapName, number>> = {
   SouthAmerica: 5,
   StraitOfGibraltar: 5,
   World: 8,
+  Middleeast: 5,
```
This needs to be moved to be between Mena and NorthAmerica.

TheGiraffe3 commented on resources/middleeast/map.json:
Rename to `manifest.json`.

TheGiraffe3 commented on resources/middleeast/mapMini.bin:
Rename to `mini_map.bin`.

TheGiraffe3 commented on resources/middleeast/mapThumb.webp:
Rename to `thumbnail.webp`.

TheGiraffe3 commented:
Also, please add screenshots of the map.

abdyar commented:
I've addressed all your feedback. Thank you


<img width="1500" height="1920" alt="map" src="https://github.com/user-attachments/assets/635ac2f4-b428-4346-a1d8-78f25bc1089a" />
<img width="638" height="806" alt="image" src="https://github.com/user-attachments/assets/34fc2e0b-96e7-4509-8d89-2231aa8a18c7" />
<img width="605" height="796" alt="image1" src="https://github.com/user-attachments/assets/65f82a78-dfba-4e35-81ef-c5cc132c7113" />

TheGiraffe3 commented:
Could you rebase this PR onto main?

abdyar commented:
Hi! I have rebased my branch on 'main' to remove the extraneous changes. It should be clean now. Please take another look when you have a moment. Thanks! (I had exams)
